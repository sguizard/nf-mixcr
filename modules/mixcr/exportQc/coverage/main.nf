process MIXCR_EXPORTQC_COVERAGE {
    tag "$meta.id"
    container 'ghcr.io/milaboratory/mixcr/mixcr:4.6.0'

    input:
    tuple val(meta), path(vdjca)
    path(library)

    output:
    path("*_exportQC_coverage.pdf")  , emit: qc_coverage_pdf
    path("*_exportQC_coverage_*.png"), emit: qc_coverage_png
    path "versions.yml"              , emit: versions

    script:
    def prefix = task.ext.prefix ?: meta.id

    """
    mixcr exportQc coverage $vdjca ${prefix}_exportQC_coverage.pdf
    mixcr exportQc coverage $vdjca ${prefix}_exportQC_coverage.png

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: meta.id
    """
    touch ${prefix}_exportQC_coverage_R1.pdf
    touch ${prefix}_exportQC_coverage_R1.png

    cat <<-END_VERSIONS > versions.yml
    MIXCR_ANALYZE:
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """
}
