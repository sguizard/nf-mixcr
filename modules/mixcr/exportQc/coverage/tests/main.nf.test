nextflow_process {

    name "Test Process MIXCR_EXPORTQC_COVERAGE"
    script "modules/mixcr/exportQc/coverage/main.nf"
    process "MIXCR_EXPORTQC_COVERAGE"

    tag "mixcr"
    tag "mixcr/exportQC/coverage"

    setup {
        run("MIXCR_ANALYZE") {
            script "../../../analyze/main.nf"
            config "../../../analyze/tests/nextflow.config"
            process {
                """
                input[0] = 
                [
                    ['id':'TCR_cat'],
                    [
                        file("${projectDir}/test-datasets/23570MA0002L01_1_20k.fastq.gz"),
                        file("${projectDir}/test-datasets/23570MA0002L01_2_20k.fastq.gz")
                    ]
                ]
                input[1] = 'generic-amplicon-with-umi'
                input[2] = file("${projectDir}/test-datasets/imgt.202312-3.sv8.json.gz")
                """
            }
        }
    }

    test("MIXCR_EXPORTQC_COVERAGE Compute coverage QC") {

        when {
            process {
                """
                input[0] = MIXCR_ANALYZE.out.vdjca
                input[1] = file("${projectDir}/test-datasets/imgt.202312-3.sv8.json.gz")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
