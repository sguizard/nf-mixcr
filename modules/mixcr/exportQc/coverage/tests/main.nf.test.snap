{
    "MIXCR_EXPORTQC_COVERAGE Compute coverage QC": {
        "content": [
            {
                "0": [
                    "[id:TCR_cat]_exportQC_coverage.pdf:md5,8cbbc6bad4714436bbb60af9df4adc20"
                ],
                "1": [
                    [
                        "[id:TCR_cat]_exportQC_coverage_R0.png:md5,5bef3e151e35e5ea576d93c15a123478",
                        "[id:TCR_cat]_exportQC_coverage_R1.png:md5,88017b1a273738d139c66d24c4787b7b",
                        "[id:TCR_cat]_exportQC_coverage_R2.png:md5,cefaa24a538f22240905e6f53e5956a5"
                    ]
                ],
                "2": [
                    "versions.yml:md5,de43f6f76d4e6e6df112b27e04aebc03"
                ],
                "qc_coverage_pdf": [
                    "[id:TCR_cat]_exportQC_coverage.pdf:md5,8cbbc6bad4714436bbb60af9df4adc20"
                ],
                "qc_coverage_png": [
                    [
                        "[id:TCR_cat]_exportQC_coverage_R0.png:md5,5bef3e151e35e5ea576d93c15a123478",
                        "[id:TCR_cat]_exportQC_coverage_R1.png:md5,88017b1a273738d139c66d24c4787b7b",
                        "[id:TCR_cat]_exportQC_coverage_R2.png:md5,cefaa24a538f22240905e6f53e5956a5"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,de43f6f76d4e6e6df112b27e04aebc03"
                ]
            }
        ],
        "timestamp": "2024-01-22T14:55:01.203987"
    }
}